packages/core/core/src/Parcel.js

```diff
@@ -94,19 +94,34 @@ export default class Parcel {
     this._initialized = true;
   }
 
-  async run(): Promise<BundleGraph> {
+  // `run()` returns `Promise<?BundleGraph>` because in watch mode it does not
```

Comment:
From a public api standpoint, what if watching were something you told Parcel to do, rather than a configuration option? conflating them both in `run` here is getting awkward.

It would be neat if `Parcel.watch()` could return something like an observable of `BundleGraph`s (or a similar summary structure containing it), since reporters have to be packages/modules on disk.